pkg/function/overrides/override.go

```diff
@@ -0,0 +1,177 @@
+package overrides
+
+import (
+	"encoding/json"
+	"fmt"
+
+	intcel "github.com/Azure/eno/internal/cel"
+	intmut "github.com/Azure/eno/internal/resource/mutation"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// mirror of type Op struct  and type jsonOp struct  in internal/resource/mutation/mutation.go
+// could pull those out
+type Override struct {
+	Path      string `json:"path"`
+	Value     any    `json:"value"`
+	Condition string `json:"condition"`
+}
+
+func (o *Override) validate() (cel.Program, error) {
+
+	if o.Path == "" {
+		return nil, fmt.Errorf("path is required")
+	}
+
+	//Not taking a dependency
+	_, err := intmut.ParsePathExpr(o.Path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse path: %w", err)
+	}
+
+	if o.Condition == "" {
+		return nil, fmt.Errorf("condition is required")
+	}
+	// Parse the expression
+	celEnv := intcel.Env
+	ast, issues := celEnv.Parse(o.Condition)
+	if issues != nil && issues.Err() != nil {
+		return nil, fmt.Errorf("failed to parse condition: %w", issues.Err())
+	}
+
+	// Type-check the expression
+	checked, issues := celEnv.Check(ast)
+	if issues != nil && issues.Err() != nil {
+		return nil, fmt.Errorf("failed to type-check condition: %w", issues.Err())
+	}
+
+	// Create the program
+	p, err := celEnv.Program(checked)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create program: %w", err)
+	}
+
+	//Value can be null which is abit wierd.
+	return p, nil
+
+}
+
+func (o *Override) Test(data map[string]interface{}) (bool, error) {
+	// Evaluate with the input data
+
+	prg, err := o.validate()
+	if err != nil {
+		return false, fmt.Errorf("failed to validate override: %w", err)
+	}
+
+	result, _, err := prg.Eval(data)
+	if err != nil {
+		return false, fmt.Errorf("failed to evaluate condition: %w", err)
+	}
+
+	// Convert result to boolean
+	if boolVal, ok := result.(types.Bool); ok {
+		return bool(boolVal), nil
+	}
+
+	return false, fmt.Errorf("condition did not evaluate to boolean, got: %T", result)
+}
+
+// AnnotateOverrides will take care of appropriatly serializng your overrides to annotations
+// merging them with others that exist
+func AnnotateOverrides(obj *unstructured.Unstructured, overrides []Override) error {
```

Comment:
Since we're only touching annotations here might as well use client.Object instead of unstructured